Replicator/Replicator.cc

```diff
@@ -1063,14 +1057,15 @@ namespace litecore { namespace repl {
         } else {
             // Either it is 3.1 client, hence collectionAware()==true,
             // or a 3.0 client and the above piece of code should have been already called.
-
-            auto [checked, err] = checkCollectionOfMsg(*request, kNotCollectionIndex);
+            auto [checked, err] = checkCollectionOfMsg(*request);
             if (err) {
                 request->respondWithError({"HTTP"_sl, 400, err});
                 return;
             }
             collectionIn = checked;
         }
+        // We don't save checkpointID by collection.
+        (void)collectionIn;
```

Comment:
Then why bother returning it?

Reply:
I don't know whether you mean why the function returns it or why in this particular call, why I use a variable to receive the returned value.
1. The reason the function returns it is that it is used in other places. Although textually it showed only once, most requests are routed by Replicator::handleConnectionMessages, where we this method is used to guarantee integrity and uniformity, whereby legacy absence of collection property is turned to index 0, etc. 
2. Second, I may choose to ignore the returned value at this place. The reason I used it here is to emphasize the protocol. The protocol requires a valid collection index, so as to allow the server save the checkpointID in place indexable by the collection index. I have the impression that SG uses the collection index. (I may be wrong.) But my current understanding is that the client is obliged to provide the collection index.